packages/@expo/metro-config/src/transform-worker/getAssets.ts

```diff
@@ -22,18 +28,67 @@ type Options = {
   publicPath: string;
 };
 
-function md5Hash(data: string[]) {
+function getMD5ForDataAsync(data: string[]) {
   if (data.length === 1) return data[0];
   const hash = crypto.createHash('md5');
   hash.update(data.join(''));
   return hash.digest('hex');
 }
 
-function assertHashedAssetData(data: AssetData): asserts data is HashedAssetData {
-  assert(
-    'fileHashes' in data,
-    'Assets must have hashed files. Ensure the expo-asset plugin is installed.'
+function getMD5ForFilePathAsync(path: string) {
+  return new Promise<string>((resolve, reject) => {
+    const output = crypto.createHash('md5');
+    const input = fs.createReadStream(path);
+    input.on('error', (err) => reject(err));
+    output.once('readable', () => resolve(output.read().toString('hex')));
+    input.pipe(output);
```

Comment:
if we don't use `pipeline` here or check for `output` errors could the read stream leak?